src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,419 @@
+<template>
+  <div class="k-tree-list">
+    <draggable
+      v-bind="draggableAttrs"
+      :disabled="disableDrag"
+      :list="internalList"
+      :group="{ name: 'k-tree-list', put: !maxLevelReached }"
+      :move="checkMove"
+      :level="level"
+      @start="onStartDrag"
+      @end="onStopDrag"
+      @change="handleChangeEvent"
+    >
+      <template #item="{ element }">
+        <div
+          :style="indentStyle"
+          :class="{
+            'non-root': level !== 0
+          }"
+          class="k-tree-item-container"
+        >
+          <KTreeItem
+            :key="`tree-item-${key}`"
+            :item="element"
+            :disabled="disableDrag"
+            :class="{
+              'has-children': hasChildren(element)
+            }"
+            @selected="handleSelection"
+          >
+            <template #item-icon>
+              <slot
+                :item="element"
+                name="item-icon"
+              >
+                <KIcon
+                  v-if="element.icon !== 'none'"
```

Comment:
No, we don't have a use case for this, but I figured we should offer it. Additionally, we can't use a prop because this field is at the item level and it could be different for each item (we may want to show an icon for some but not others).